### DISCUSSION TITLE
Adding manufacturer specific attributes

### DISCUSSION THREAD

**Discussion by inventorofthingies** [zigpy/zigpy#1276] [General]
Please can someone point me to docs (or a good example) for creating custom clusters (assuming that is what I need).
I'm looking to add a new quirk and need to add manufacturer specific attributes (that can be read and written). I've got the signature to match for a quirk and then I think I need to create a class inheriting from CustomCluster and put that in for the appropriate replacement.

Adding a bit more context -- I want to use the inputs from a Salus SR600 which needs a manufacturer specific attribute to configure how the switch inputs behave, and to read the value of them.
https://github.com/dresden-elektronik/deconz-rest-plugin/issues/2109#issuecomment-560022043

I suspect I need to work out how to create an appropriate event in HA too, but one step at a time!

**Comment by Hedda** [comment]
Custom cluster definitions are listed in  README.md file in the ZHA Device Handlers repository, other than that there is only existing quirks:

https://github.com/zigpy/zha-device-handlers/blob/dev/README.md

@inventorofthingies a tip is that you probably get more attention posting questions about quirks in the ZHA Device Handlers repository, especially if you create a new "Device support request" for that specific devices and then ask specific questions with references in that issue:

 
https://github.com/zigpy/zha-device-handlers/issues

Alternativly start new discussions as seperate "Q&A" with specific question in each as those then help work as a FAQ for others in the future:

https://github.com/zigpy/zha-device-handlers/discussions
